Sources/Workspace/Workspace.swift

```diff
@@ -366,15 +377,19 @@ public class Workspace {
                 manifestVersion: toolsVersion.manifestVersion)
 
             guard manifest.name == packageName else {
-                throw WorkspaceOperationError.mismatchingDestinationPackage(
-                    path: destination, destPackage: manifest.name, expectedPackage: packageName)
+                engine.emit(WorkspaceError.mismatchingDestinationPackage(
+                    path: destination,
+                    destinationPackage: manifest.name,
+                    expectedPackage: packageName))
+                return
             }
+            
             // Emit warnings for branch and revision, if they're present.
             if let checkoutBranch = checkoutBranch {
-                delegate.warning(message: "not checking out branch '\(checkoutBranch)' for dependency '\(packageName)'")
+                engine.emit(warning: "not checking out branch '\(checkoutBranch)' for dependency '\(packageName)'")
```

Comment:
This overload does not exist yet. But I was planning on something like: `emit.(WorkspaceWarning.unusedCheckoutBranchArgument)` and define the text in the associated diagnostic ID.